docs/configuration.md

```diff
@@ -326,4 +326,209 @@ deployment_config = DeploymentConfig(
        'docker_repository': 'your_repository',
        'gcp_project_id': '{project_id}',
        'dags_bucket': '{dags_bucket}'}})
-```
\ No newline at end of file
+```
+
+
+# Class-based configuration [experimental]
+
+Bigflow 1.3 introduces new class-based approach for keeping configurations.
+New API is optional and there is no need to migrate existing code from
+`biglfow.Configuration`. However it betters plays with typehinting, enables
+autocompletion in IDEs and gives more flexibility.
+
+Each configuration is declared as a subclass of `bigflow.konfig.Konfig`.
+Properties are declared as attributes, optionally with type hints.
+Classes may inherit from each other and share common properties.  Standard
+python inheritance rules are used to resolve overriden properties.
+Custom methods may be added to classes, however it is not recommended.
+
+```python
+import bigflow.konfig
+
+class CommonConf(bigflow.konfig.Konfig):
+    project: str                    # type hint
+    region: str = 'europe-west1'    # type hint + default value
+    machine_type = 'n2-standard-4'  # only value (type is `typing.Any`)
+    num_workers = 4                 # not only strings are allowed
+
+class DevConf(CommonConf):
+    project = "dev-project"
+
+class ProdConf(CommonConf):
+    project = "prod-project"
+    num_workers = 20
+```
+
+Config which corresponds to the current environment may be created by function `resolve_konfig`.
+It takes dictionary in form of {"environment" -> "config class"} and returns instance of the config.
+
+```python
+config = bigflow.konfig.resolve_konfig(
```

Comment:
Cant we automatize this mapping somehow? It adds additional step to make and is error prone. Can't we require from every Konfig instance to contain `env` variable?

Reply:
Yes, we can, initial "versions" of `konfig.py` had ability to use list of konfig classes instead of dictionary.
Also there was method for auto-scan all subclasses (eg `config = CommonConf.resolve(default='dev')`.

But now it can be only dict (by design).  Mosty for making list of envs/configs explicit (you immediately see what environments you have in you project).

Reply:
I agree with @anjensan. Having a single way of resolving configuration is better than having 2, and doing it through the dict mechanism is easier to understand and manipulate.